app/controllers/reservations_controller.rb

```diff
@@ -138,6 +138,17 @@ def show
     @reservation  = Reservation.find(params[:id])
 
     raise ActiveRecord::RecordNotFound if @reservation != @order_detail.reservation
+
+    respond_to do |format|
+      format.html
+
+      format.ics do
+        calendar = ReservationCalendar.new(@reservation, request.host)
+        send_data(calendar.to_ical,
+                  type: 'text/calendar', disposition: 'attachment',
+                  filename: "reservation_cal_#{@reservation.id}.ics")
```

Comment:
An order might have multiple reservations, but an order detail can only have one reservation. We use `OrderDetail#to_s` or `OrderDetail#order_number` (they're aliases) in quite a few places.